### PR TITLE
➕ Added Crane (Demo)

### PR DESCRIPTION
**Pull request by kulothunganug**
- Added crane demo (Material Study)
Screenshot 2020-11-28 10:22:57
- Added tooltip_radius attr
- transfered `MagicBehavior` factory.register from `magic_behavior.py` to `factory_registers.py`

### CODE REVIEW

File: demos/crane/libs/kv/root_screen.kv [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    		Image:
+    			source: f"{environ['CRANE_ROOT']}/assets/images/logo.png"
+    			size_hint: (None, None)
+    			size: dp(40), dp(40)
+
+            MDBoxLayout:
+
+        		CraneNavigationLabel:
+        			text: "FLY"
+                    state: "down"
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Indentation error

File: demos/crane/libs/kv/root_screen.kv [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        			text: "EAT"
+        			pos_hint: {"center_y": .5}
+        			on_press:
+                        scr.transition.direction = "left"
+                        scr.current = "eat"
+                Widget:
+
+        ScreenManager:
+            id: scr
+            CraneFlyScreen:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Please add one indent

File: demos/crane/libs/kv/root_screen.kv [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                name: "fly"
+            CraneSleepScreen:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Please add one indent

File: demos/crane/libs/kv/root_screen.kv [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                name: "sleep"
+            CraneEatScreen:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Please add one indent

File: demos/crane/libs/kv/root_screen.kv [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	line_color: (0, 0, 0, 0)
+	foreground_color: gch("#ffffff")
+    font_name: f"{environ['CRANE_ROOT']}/assets/fonts/Raleway-Medium.ttf"
+
+<CraneListItem>
+    size_hint_y: None
+    padding: dp(20)
+    spacing: dp(10)
+    height: dp(80)
+    canvas:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Please add one indent

File: demos/crane/libs/kv/eat_screen.kv [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+			canvas.before:
+				Color:
+					rgba: root.theme_cls.bg_normal
+				RoundedRectangle:
+					size: self.size
+					pos: self.pos
+					radius: [20, 20, 0, 0]
+
+			ScrollView:
+				MDList:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/eat_screen.kv:40]: Please add one indent

File: demos/crane/libs/kv/eat_screen.kv [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+					MDBoxLayout:
+						adaptive_height: True
+						padding: dp(15)
+
+						MDLabel:
+							text: "Explore restaurants by destination"
+				            font_style: "Subtitle2"
+				            theme_text_color: "Hint"
+				            adaptive_height: True
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/eat_screen.kv]: Indentation error

File: demos/crane/libs/kv/fly_screen.kv [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+			canvas.before:
+				Color:
+					rgba: root.theme_cls.bg_normal
+				RoundedRectangle:
+					size: self.size
+					pos: self.pos
+					radius: [20, 20, 0, 0]
+
+			ScrollView:
+				MDList:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/fly_screen.kv:40]: Please add one indent

File: demos/crane/libs/kv/fly_screen.kv [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+					MDBoxLayout:
+						adaptive_height: True
+						padding: dp(15)
+
+						MDLabel:
+							text: "Explore flights by destinations"
+				            font_style: "Subtitle2"
+				            theme_text_color: "Hint"
+				            adaptive_height: True
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/fly_screen.kv]: Indentation error

File: demos/crane/libs/kv/root_screen.kv [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    MDBoxLayout:
+        orientation: "vertical"
+        md_bg_color: gch("#5D1049")
+
+    	MDBoxLayout:
+            size_hint_y: None
+            height: root.theme_cls.standard_increment
+    		md_bg_color: gch("#5D1049")
+    		padding: dp(10)
+    		spacing: dp(10)
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Indentation error

File: demos/crane/libs/kv/root_screen.kv (hunk 1 above)
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Indentation error

File: demos/crane/libs/kv/root_screen.kv [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+            MDBoxLayout:
+
+        		CraneNavigationLabel:
+        			text: "FLY"
+                    state: "down"
+        			pos_hint: {"center_y": .5}
+        			on_press:
+                        scr.transition.direction = "right"
+                        scr.current = "fly"
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/root_screen.kv]: Indentation error

File: demos/crane/libs/kv/sleep_screen.kv [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+					MDBoxLayout:
+						adaptive_height: True
+						padding: dp(15)
+
+						MDLabel:
+							text: "Explore properties by destination"
+				            font_style: "Subtitle2"
+				            theme_text_color: "Hint"
+				            adaptive_height: True
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/sleep_screen.kv]: Indentation error

File: demos/crane/libs/kv/sleep_screen.kv [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+			canvas.before:
+				Color:
+					rgba: root.theme_cls.bg_normal
+				RoundedRectangle:
+					size: self.size
+					pos: self.pos
+					radius: [20, 20, 0, 0]
+
+			ScrollView:
+				MDList:
```
Comment (HeaTTheatR) [inline demos/crane/libs/kv/sleep_screen.kv:37]: Please add one indent

File: demos/crane/main.py [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+KV = """
+#:import FadeTransition kivy.uix.screenmanager.FadeTransition
+#:import CraneRootScreen libs.baseclass.root_screen.CraneRootScreen
+
+ScreenManager:
+    transition: FadeTransition()
+
+    CraneRootScreen:
+        name: "crane root screen"
+
```
Comment (HeaTTheatR) [inline demos/crane/main.py]: Please remove one indent

**Review by HeaTTheatR — CHANGES_REQUESTED** [review]

**Comment by kulothunganug** [comment]
@HeaTTheatR The Indentation Errors are created by the mixture of tabs and spaces. I fixed those ✔️

**Pull request merged**